File: queue/queue.py
class Queue:
    def __init__(self, size):
        self.contents = [None] * size
        self.front = 0
        self.rear = -1
        self.count = 0
        self.size = size

    def enQueue(self, element):
        if self.count == self.size:
            print("Queue is full")
            return
        self.rear = (self.rear + 1) % self.size
        self.contents[self.rear] = element
        self.count += 1

    def deQueue(self):
        if self.count == 0:
            print("Queue is empty")
            return -1000
        data = self.contents[self.front]
        self.front = (self.front + 1) % self.size
        self.count -= 1
        return data

    def display(self):
        if self.count == 0:
            print(" {}")
            return
        j = self.front
        for i in range(self.count):
            print(self.contents[j], end=" ")
            j = (j + 1) % self.size
        print()

def printMenu():
    print("Choice 1 : Enter element into Queue")
    print("Choice 2 : Delete element from Queue")
    print("Choice 3 : Display")
    print("Any other choice : Exit")

if __name__ == "__main__":
    queue = Queue(5)

    while True:
        printMenu()
        choice = int(input("Enter your choice: "))
        if choice == 1:
            data = int(input("Enter the element to be inserted/entered: "))
            queue.enQueue(data)
        elif choice == 2:
            data = queue.deQueue()
            if data != -1000:
                print("The deleted element is", data)
        elif choice == 3:
            print("The contents of the queue are", end=" ")
            queue.display()
        else:
            break

File: queue/queue.c
#include <stdio.h>
#include <string.h>

struct queue {
    int contents[5];
    int front;
    int rear;
    int count;
};

void printMenu() {
    printf("Choice 1 : Enter element into Queue\nChoice 2 : Delete element from Queue\nChoice 3 : Display\nAny other choice : Exit\n");
}

void initQueue(struct queue* q) {
    q->front = 0;
    q->rear = -1;
    q->count = 0;
}

void enQueue(struct queue* q, int element) {
    if (q->count == 5) {
        printf("Queue is full\n");
        return;
    }
    q->rear = (q->rear + 1) % 5;
    q->contents[q->rear] = element;
    q->count++;
}

int deQueue(struct queue* q) {
    if (q->count == 0) {
        printf("Queue is empty\n");
        return -1000;
    }
    int d = q->contents[q->front];
    q->front = (q->front + 1) % 5;
    q->count--;
    return d;
}

void display(struct queue* q) {
    if (q->count == 0) {
        printf(" {}");
        return;
    }
    int j = q->front;
    for (int i = 0; i < q->count; i++) {
        printf(" %d", q->contents[j]);
        j = (j + 1) % 5;
    }
}

int main() {
    struct queue p;
    int data, ch, data1;
    initQueue(&p);
    do {
        printMenu();
        printf("Enter your choice\n");
        scanf("%d", &ch);
        switch (ch) {
        case 1:
            printf("Enter the element to be inserted/entered\n");
            scanf("%d", &data);
            enQueue(&p, data);
            break;
        case 2:
            data1 = deQueue(&p);
            if (data1 != -1000)
                printf("The deleted element is %d\n", data1);
            break;
        case 3:
            printf("The contents of the queue are");
            display(&p);
            printf("\n");
            break;
        default:
            return 0;
        }
    } while (1);
    return 0;
}
